sub-packages/bionemo-moco/src/bionemo/moco/schedules/inference_time_schedules.py

```diff
@@ -458,3 +457,160 @@ def generate_schedule(
+        """
+        # Detach and set requires_grad=True to compute gradients with respect to x
+        x = x.detach().requires_grad_(True)
+        
+        # Generate a random vector from the Rademacher distribution ({-1, 1})
+        epsilon = (torch.randint_like(x, 0, 2) * 2 - 1).to(x.dtype)
+        
+        # Trtansform input
+        model_input = torch.cat([x, t], dim=-1)
+        v = model(model_input)
+        
+        # Calculate the Jacobian-vector product (JVP)
+        # This computes (d(v)/d(x)) * epsilon without forming the full Jacobian
+        jvp = torch.autograd.grad(outputs=v, inputs=x, grad_outputs=epsilon, create_graph=False)[0]
+        
+        # The divergence is estimated by the dot product: (JVP^T * epsilon)
+        # which is equivalent to sum(jvp * epsilon) over the feature dimension.
+        divergence_est = (jvp * epsilon).view(jvp.shape[0], -1).sum(dim=-1)
+        
+        return divergence_est
+
+    def _calculate_entropy_rate(self, t_val: float) -> float:
+        """Calculates the mean entropy rate at a specific time t."""
+        t_batch = torch.full((self.batch_size, 1), t_val, device=self.device)
+        x_0 = self.x_0_sampler(self.batch_size).to(self.device)
+        x_1 = self.x_1_sampler(self.batch_size).to(self.device)
+        
+        # Calculate one step
+        x_t = (1 - t_val) * x_0 + t_val * x_1
+        
+        # No need for torch.no_grad() here as it's handled in the main loop
+        div = self._hutchinson_divergence(t_batch, x_t)
+        return div.mean().item()
+
+    def generate_schedule(
+        self, 
+        nsteps: Optional[int] = None, 
+        device: Optional[Union[str, torch.device]] = None
+    ) -> Tensor:
+        """Generates the entropic time schedule."""
+        dev = device if device is not None else self.device
+        n_final_steps = nsteps if nsteps is not None else self.nsteps
+        
+        # Calculating entropic profile over {self.n_schedule_points}...
+        standard_time = torch.linspace(0, 1, self.n_schedule_points, device=dev)
+        entropy_rates = torch.empty(self.n_schedule_points, device=dev)
+        
+        for i, t_val in enumerate(standard_time):
+            entropy_rates[i] = self._calculate_entropy_rate(t_val.item())
+
+        entropy_rates = entropy_rates.clamp(min=0)
+        cumulative_entropy = torch.cumsum(entropy_rates, dim=0)
+        
+        if cumulative_entropy[-1] > 1e-6:
+            cumulative_entropy = cumulative_entropy / cumulative_entropy[-1]
+        
+        num_interp_points = n_final_steps + 1 if not self.inclusive_end else n_final_steps
+
+        # Uniform_time is just a projection of the interpolated points
+        uniform_time = torch.linspace(0, 1, num_interp_points, device="cpu")
+        
+        import numpy as np 
+        entropic_schedule_np = np.interp(
+            uniform_time.numpy(), cumulative_entropy.cpu().numpy(), standard_time.cpu().numpy()
+        )
```

Comment:
Can this be entropic_schedule_torch = torch.interp(uniform_time, cumulative_entropy, standard_time)